### PR TITLE
Fix #1804 Update gradle version

### LINKED ISSUE #1804: Update gradle version.
**Describe the feature**

Gradle version should be updated to the latest version available and all requisite changes should be made to the scripts. 

**Additional Context**

I would like to work on this.

### PR DESCRIPTION
**Pull request by yashk2000**
Fixes #1804 

Updated gradle to the latest version, 3.6.0

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -44,6 +44,7 @@ android {
     resValue("string", "app_search_string", "Search Kiwix")
     versionCode = generateVersionCode()
     versionName = generateVersionName()
+    ndkVersion  = "21.0.6113669"
```
Comment (macgills) [inline app/build.gradle.kts]: we shouldn't be doing this

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Sun Jun 16 15:57:45 CDT 2019
+#Tue Feb 25 11:46:45 IST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
```
Comment (macgills) [inline gradle/wrapper/gradle-wrapper.properties]: This is what I wanted to be upgraded. The latest release is 6.2.1

File: custom/build.gradle.kts [hunk 3]
```diff
@@ -17,6 +17,7 @@ plugins.apply(KiwixConfigurationPlugin::class)
 android {
   defaultConfig {
     applicationId = "org.kiwix"
+    ndkVersion  = "21.0.6113669"
```
Comment (macgills) [inline custom/build.gradle.kts]: remove this

File: buildSrc/build.gradle.kts [hunk 4]
```diff
@@ -11,7 +11,7 @@ repositories {
 }
 
 dependencies {
-  implementation("com.android.tools.build:gradle:3.5.3")
+  implementation("com.android.tools.build:gradle:3.6.0")
```
Comment (macgills) [inline buildSrc/build.gradle.kts]: I am probably going to make a PR upgrading the build tools now as the AS 3.6 update dropped just as I was done work yesterday

File: custom/build.gradle.kts (hunk 3 above)
Comment (yashk2000) [reply custom/build.gradle.kts]: I'll try again after removing this, but the build was failing unless I specifed this.

File: buildSrc/build.gradle.kts (hunk 4 above)
Comment (yashk2000) [reply buildSrc/build.gradle.kts]: OK. I'll revert that.

**Comment by yashk2000** [comment]
@macgills I reverted all changes except those to `gradle-wrapper.properties`.
It caused a build error. 

```
A problem occurred evaluating script.
> Cannot set the value of read-only property 'sourceDirectories' for task ':app:jacocoInstrumentationTestReport' of type org.gradle.testing.jacoco.tasks.JacocoReport.
```

We could search for a workaround for this, but since `com.android.tools.build:gradle` also has to be updated sooner or later, should we just combine the changes?

**Comment by macgills** [comment]
@yashk2000 this is the updates to our gradle tasks I talked about on the ticket, look at `jacoco.gradle`
image

**Comment by yashk2000** [comment]
@macgills I'm not able to resolve this error for `executionData`.

```
executionData.setFrom(fileTree(execution))
executionData.from = file(execution)
executionData.from = fileTree(execution)
```

I tried the above, but I'm still getting this error:

```A problem occurred configuring project ':app'.

> groovy.lang.GroovyRuntimeException: Cannot set the value of read-only property 'executionData' for task ':app:jacocoTestDebugUnitTestReport' of type org.gradle.testing.jacoco.tasks.JacocoReport.
```

**Comment by macgills** [comment]
@yashk2000 
image
does that work?

**Comment by yashk2000** [comment]
@macgills I tried that. It didn't work.

**Comment by macgills** [comment]
@yashk2000 maybe this https://stackoverflow.com/a/56181389/12225251 ? You have set the gradle version to ` 6.2.1`?

**Comment by macgills** [comment]
ah I found the issue, a plugin we use for jacoco has not been upgraded https://github.com/arturdm/jacoco-android-gradle-plugin/issues/72 

A kind member of the open source community has forked it to allow 6.0 support.

https://plugins.gradle.org/plugin/com.hiya.jacoco-android

to apply this go to `buildSrc/build.gradle.kts` and update
`  implementation("com.dicedmelon.gradle:jacoco-android:0.1.4")`
with
`"com.hiya:jacoco-android:0.2"`

And in AllProjectConfigurer replace
`    target.plugins.apply("jacoco-android")`
with
`    target.plugins.apply("com.hiya.jacoco-android")`

**Comment by yashk2000** [comment]
> ah I found the issue, a plugin we use for jacoco has not been upgraded [arturdm/jacoco-android-gradle-plugin#72](https://github.com/arturdm/jacoco-android-gradle-plugin/issues/72)
> 
> A kind member of the open source community has forked it to allow 6.0 support.
> 
> https://plugins.gradle.org/plugin/com.hiya.jacoco-android
> 
> to apply this go to `buildSrc/build.gradle.kts` and update
> ` implementation("com.dicedmelon.gradle:jacoco-android:0.1.4")`
> with
> `"com.hiya:jacoco-android:0.2"`
> 
> And in AllProjectConfigurer replace
> ` target.plugins.apply("jacoco-android")`
> with
> ` target.plugins.apply("com.hiya.jacoco-android")`

@macgills great! I was also thinking of suggesting doing this. I'll try it out and update the pr.

**Comment by yashk2000** [comment]
@macgills this works now. Is it fine?

**Pull request merged**